posts/exploring-iot-infrastructure-support-in-aws-part1.md

```diff
@@ -0,0 +1,211 @@
+2. Development options (SDKs, Software, Hardware, 3rd party)
+3. An brief introduction to a proof concept and snippets of code
+4. Inital impressions and comparisons with other cloud providers
+
+# What will I be covering in part 2?
+1. Introduction to a the POC and full discussion
+2. Further expansion on chosen SDKs, software and hardware
+3. What I liked, what I did not like during developement
+4. Evaluation of AWS services and the POC
+5. Links and Resources
+
+![ESP8266](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/07/aws_iot_esp_device.jpeg)
+
+# IoT Configuration
+AWS provides a hub like [infrastructure](https://docs.aws.amazon.com/iot/latest/developerguide/what-is-aws-iot.html) which incorporates IoT devices as "[Things](https://docs.aws.amazon.com/iot/latest/developerguide/iot-thing-management.html)" in the network. This network is based heavily on "[Device Shadows](https://docs.aws.amazon.com/iot/latest/developerguide/iot-device-shadows.html)" which store device state and the MQTT network protocol for sending and receiving messages. A shadow is simply a JSON document containing specified, expected and delta of device state. These devices are categorised into either control/management devices called a “Greengrass Core” or lower powered device called a “Thing” which interacts with “Cores” within a Greengrass group using the AWS SDK. The terminology is somewhat confusing with “Greengrass cores”, “Greengrass groups”, “IoT core”, “Things” etc. A Greengrass core is not needed to setup IoT devices but gives additional benefits. The diagram below will hopefully explain the structure of AWS IoT offerings. Things are split across both Greengrass groups and normal IoT groups. To setup a Greengrass group you must select a "Thing" to act as the core. This thing should be setup to run the Greengrass core software.
+
+![Core AWS IoT infrastructure](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/07/aws_iot_overview.png)
+
+The lower powered devices acting at the edge usually have a CPU less than 1Ghz, an interface with sensors and run in an environment with limited connectivity.  The “Greengrass Core” has a custom “Greengrass” daemon and a number of software builds depending on the platform. There is support for Raspberry Pi, EC2, x86 and Arm. The lower powered “Thing” devices can use [AWS RTOS](https://aws.amazon.com/freertos/), AWS SDKs or the REST API to interact with IoT services. 
+
+Communication is supported over MQTT and HTTP protocols with additional Websocket support. Both Greengrass Cores, groups and IoT things are managed through the same UI and appear to be very similar with the exception of additional interfaces for Greengrass Lambda functions and setting up groups in Greengrass.
+
+## AWS IoT Product Overview:
+
+The following table gives my opinions on the AWS IoT services, SDK and a summary of the intended uses.
+Product | Summary | SDK/HW | Verdict
```

Comment:
This table doesn't display correctly, see https://github.com/BenEdridge/blog-usa/blob/master/posts/exploring-iot-infrastructure-support-in-aws-part1.md